generator/src/lib.rs

```diff
@@ -420,6 +427,31 @@ pub fn generate_phf(
     }
 }
 
+fn normalise_name(s: &str, codepoint: char) -> String {
+    let mut normalised = String::new();
+    let bytes = s.as_bytes();
+    for (i, c) in bytes.iter().copied().enumerate() {
+        if c.is_ascii_whitespace() || c == b'_' {
+            continue;
+        }
+        if codepoint != '\u{1180}' // HANGUL JUNGSEONG O-E
+            && c == b'-'
+            && bytes.get(i - 1).map_or(false, u8::is_ascii_alphanumeric)
+            && bytes.get(i + 1).map_or(false, u8::is_ascii_alphanumeric)
+        {
+            continue;
+        }
+        assert!(
+            c.is_ascii_alphanumeric() || c == b'-',
+            "{:?} isn't a valid character for a Unicode name",
+            c as char
+        );
+        normalised.push(c.to_ascii_uppercase() as char);
+    }
+
+    normalised
+}
```

Comment:
```suggestion
/// Returns a name that can be used for matching, as per
/// [UAX#44](https://www.unicode.org/reports/tr44/tr44-34.html#Matching_Names)
fn normalise_name(s: &str, codepoint: char) -> String {
    let mut normalised = String::new();
    let bytes = s.as_bytes();
    for (i, c) in bytes.iter().copied().enumerate() {
        // "Ignore case, whitespace, underscore ('_'), and [...]"
        let c = c.to_ascii_uppercase();
        if c.is_ascii_whitespace() || c == b'_' {
            continue;
        }

        // "[...] all medial hyphens except the hyphen in U+1180 HANGUL JUNGSEONG O-E."
        if codepoint != '\u{1180}' // HANGUL JUNGSEONG O-E
            && c == b'-'
            && bytes.get(i - 1).map_or(false, u8::is_ascii_alphanumeric)
            && bytes.get(i + 1).map_or(false, u8::is_ascii_alphanumeric)
        {
            continue;
        }

        assert!(
            c.is_ascii_alphanumeric() || c == b'-',
            "{:?} isn't a valid character for a Unicode name",
            c as char
        );
        normalised.push(c as char);
    }

    normalised
}
```

makes it more readable IMO

Reply:
Agreed, I'll add that